src/CodeGen_LLVM.cpp

```diff
@@ -474,6 +475,8 @@ MangledNames get_mangled_names(const LoweredFunc &f, const Target &target) {
 }  // namespace
 
 std::unique_ptr<llvm::Module> CodeGen_LLVM::compile(const Module &input) {
+    input_module = &input;
```

Comment:
a Module is already a shared pointer - why does CodeGen_LLVM take a pointer to one?